### PR TITLE
feat(torch): add inference for torch detection models

### PR DESCRIPTION
**Pull request by BynaryCobweb**
Supports detection models that are available in torchvision: Faster
RCNN, RetinaNet

### CODE REVIEW

**Review by fantes — APPROVED** [review]

File: src/backends/torch/torchlib.cc [hunk 1]
```diff
@@ -248,6 +248,10 @@ namespace dd
       {
         _seq_training = true;
       }
+    else if (_template == "fasterrcnn" || _template == "retinanet")
+      {
+        // bbox = true; // ?
```
Comment (beniz) [inline src/backends/torch/torchlib.cc:253]: We could force it automatically yes.

File: src/backends/torch/torchlib.cc [hunk 2]
```diff
@@ -967,6 +980,25 @@ namespace dd
           forward_method = ad_mllib.get("forward_method").get<std::string>();
       }
 
+    if (output_params.has("bbox") && output_params.get("bbox").get<bool>())
+      {
+        bbox = true;
+      }
+    if (output_params.has("confidence_threshold"))
```
Comment (beniz) [inline src/backends/torch/torchlib.cc:987]: Interesting, we were not using it for classification models then! Then we should/could fix it here or in another tiny PR for classification models ?

File: src/backends/torch/torchlib.cc [hunk 3]
```diff
@@ -1117,7 +1149,88 @@ namespace dd
             if (_module._native != nullptr)
               output = _module._native->cleanup_output(output);
 
-            if (_classification)
+            if (bbox)
+              {
+                // Supporting only Faster RCNN format at the moment.
```
Comment (beniz) [inline src/backends/torch/torchlib.cc:1154]: Since RetinaNet is mentioned in the PR, this only applies to Faster-RCNN, correct ?

Comment (BynaryCobweb) [reply src/backends/torch/torchlib.cc:1154]: I forgot to change the comment, this works for RetinaNet as well.
There are two pure detection models in torchvision: Faster RCNN and RetinaNet, and they have the same output format.

File: src/backends/torch/torchlib.cc (hunk 1 above)
Comment (BynaryCobweb) [reply src/backends/torch/torchlib.cc:253]: Do we need to pass it to the output parameters at predict time then?

File: src/backends/torch/torchlib.cc (hunk 2 above)
Comment (BynaryCobweb) [reply src/backends/torch/torchlib.cc:990]: Will fix in another PR

File: src/backends/torch/torchlib.cc (hunk 1 above)
Comment (beniz) [reply src/backends/torch/torchlib.cc]: Normally yes, but if we hardcode it, it'll provide some slack. There's no use not setting it up actually.

File: src/backends/torch/torchlib.cc (hunk 2 above)
Comment (beniz) [reply src/backends/torch/torchlib.cc:990]: Let's not forget!

**Review by beniz — APPROVED** [review]

**Comment by beniz** [comment]
@BynaryCobweb could you add to the PR description the steps to get a model and execute an inference ? It's always useful.

**Pull request merged**